src/psyclone/gocean1p0.py

```diff
@@ -1091,17 +1089,14 @@ def gen_ocl(self, parent):
                     arguments.append(arg.dereference(garg.name))
                 else:
                     arguments.append(garg.name+"%grid%"+arg.name+"_device")
-        sub_name = self._name_space_manager.create_name(
-            root_name=self.name+"_set_args", context=self.name+"ArgSetter",
-            label=self.name+"_set_args")
+        sub_name = self.root.symbol_table.name_from_tag(
```

Comment:
If the plan is to remove `name_from_tag` eventually then I would not expect to see it in `gocean1p0.py`

Reply:
Done, now I use `lookup_with_tag` and make sure the _set_args subroutines are declared before being used.